newrelic/resource_newrelic_synthetics_label.go

```diff
@@ -0,0 +1,149 @@
+	return &schema.Resource{
+		Create: resourceNewRelicSyntheticsLabelCreate,
+		Read:   resourceNewRelicSyntheticsLabelRead,
+		Delete: resourceNewRelicSyntheticsLabelDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"monitor_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				// Case fold this attribute when diffing
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					return strings.EqualFold(old, new)
+				},
+			},
+			"value": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				// Case fold this attribute when diffing
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					return strings.EqualFold(old, new)
+				},
+			},
+			"href": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceNewRelicSyntheticsLabelCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ProviderConfig).NewClient
+	monitorID := d.Get("monitor_id").(string)
+	label := expandSyntheticsLabel(d)
+
+	log.Printf("[INFO] Creating New Relic Synthetics label %s:%s", label.Type, label.Value)
+
+	err := client.Synthetics.AddMonitorLabel(monitorID, label.Type, label.Value)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(strings.Join([]string{monitorID, label.Type, label.Value}, ":"))
+
+	return nil
+}
+
+func resourceNewRelicSyntheticsLabelRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ProviderConfig).NewClient
+
+	log.Printf("[INFO] Reading New Relic Synthetics label %s", d.Id())
+
+	ids := strings.Split(d.Id(), ":")
+	monitorID := ids[0]
+	labelType := ids[1]
+	value := ids[2]
+
+	_, err := client.Synthetics.GetMonitor(monitorID)
+	if err != nil {
+		if _, ok := err.(*errors.NotFound); ok {
+			d.SetId("")
+			return nil
+		}
+
+		return err
+	}
+
+	labels, err := client.Synthetics.GetMonitorLabels(monitorID)
+	if err != nil {
+		return err
+	}
+
+	var label *synthetics.MonitorLabel
+	for _, l := range labels {
+		if !strings.EqualFold(l.Type, labelType) || !strings.EqualFold(l.Value, value) {
+			continue
+		}
+
+		label = l
+	}
+
+	if label == nil {
+		d.SetId("")
+		return nil
+	}
+
+	return flattenSyntheticsLabel(label, d)
+}
+
+func resourceNewRelicSyntheticsLabelDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ProviderConfig).NewClient
+
+	ids := strings.Split(d.Id(), ":")
+	monitorID := ids[0]
+	labelType := ids[1]
+	value := ids[2]
+
+	log.Printf("[INFO] Deleting New Relic alert condition %s", d.Id())
+
+	err := client.Synthetics.DeleteMonitorLabel(monitorID, labelType, value)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func expandSyntheticsLabel(d *schema.ResourceData) *synthetics.MonitorLabel {
```

Comment:
Not a blocker, but could potentially put these into the structures_{resource}.go pattern and add some a couple unit tests for expand and flatten.

Reply:
however, I see that they are the top-level scenario that we talked about before, so maybe no big deal.

Reply:
I left them here since there was no data source that needed to share them.

Reply:
This makes sense though, I'll make it so

Reply:
@sanderblue that's right, seems like there is no good way to unit test at the top level since the expand and flatten functions depend on an instace of `schema.ResourceData`.  I will leave this as-is after all.